### DISCUSSION TITLE
Does Bear recognize the Cray compiler wrappers?

### DISCUSSION THREAD

**Discussion by dkokron** [rizsotto/Bear#577] [General]
Does Bear recognize the Cray compiler wrappers (ftn, cc, CC)?

**Comment by rizsotto** [comment]
Not out of the box, but you can add "compilers to recognise" in the config file.

**Reply by Freed-Wu** [reply]
How about adding more compilers?

**Reply by rizsotto** [reply]
sure. I might need the followings:

- point out documentation of the compiler (which details the flags and parameters it takes).
- maybe document how to install this in a ubuntu machine.
- maybe document how the installed executable is named. (packagers sometime rename the original executable name in order to support multiple version installed on the same system.)

also, PRs are welcome!

**Reply by Freed-Wu** [reply]
I can provide some information about TI compiler for TI DSP.

TI compiler for C6000, C2000, MSP 430, arm and arm based clang
use name:

cl6x, cl2000, cl430, armcl, tiarmclang.

https://www.ti.com/tool/download/CCSTUDIO

will provide all

Or install standalone like

https://www.ti.com/tool/download/C6000-CGT/7.4.24

**Reply by rizsotto** [reply]
just an update on this... there is new version where the compiler support had more attention and cleaned up the codebase.

The [compilers](https://github.com/rizsotto/Bear/tree/master/bear/src/semantic/interpreters/compilers) module has good collection of compilers that the tool is recognizing. If you have the capacity to contribute, I would be happy to check out a PR on this.
